plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/MonitorInfo.java

```diff
@@ -0,0 +1,174 @@
+package eu.arrowhead.core.plantdescriptionengine;
+
+import eu.arrowhead.core.plantdescriptionengine.utils.Metadata;
+import se.arkalix.ServiceRecord;
+import se.arkalix.SystemRecord;
+import se.arkalix.codec.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Object used for keeping track of inventory data of monitorable systems.
+ */
+public class MonitorInfo {
+
+    private final Map<String, Bundle> infoBundles = new ConcurrentHashMap<>();
+
+    /**
+     * @param service An Arrowhead Framework service.
+     * @return A unique identifier for the given service.
+     */
+    private String toKey(final ServiceRecord service) {
+
+        Map<String, String> metadata = service.provider().metadata();
+        String result = "name=" + service.provider().name() +
+            ",serviceUri=" + service.uri();
+
+        if (metadata != null) {
+            result += ",metadata=" + Metadata.toString(metadata);
+        }
+
+        return result + "}";
```

Comment:
Don't find the opening brace, maybe missing.

Reply:
Fixed!